atomicwrites/__init__.py

```diff
@@ -20,6 +20,8 @@ def _path_to_unicode(x):
 
 if sys.platform != 'win32':
     def _replace_atomic(src, dst):
+        if os.path.exists(dst):
+            os.chmod(src, os.stat(dst).st_mode)
```

Comment:
Could you store the mode somewhere, and then use `chmod` after the rename, so we don't have to check whether the file exists?